packages/collector/Collector_Content.php

```diff
@@ -0,0 +1,18 @@
+<?php
+function collector_iterate_directory($path, $prefix, $callback)
+{
+	$handle = opendir($path);
+
+	while($entry = readdir($handle))
+	{
+		if($entry === '.' || $entry === '..' || substr($entry, 0, 1) === '.')
+		{
+			continue;
+		}
+
+		$realPath = realpath($path . '/' . $entry);
+		$packPath = substr($realPath, strlen($prefix));
+
+		$callback($realPath, $packPath);
```

Comment:
Could that be a generator that yields the computed paths instead of calling a callback? Callbacks in PHP seem like a bit less natural choice than in, say, JS. Happy to discuss.